client/src/push-content.ts

```diff
@@ -9,11 +9,179 @@ export enum Tag {
   candidate = 'Release Candidate'
 }
 
+export enum DocumentsToOpen {
+  all = 'All Content',
+  modified = 'Modified Content'
+}
+
 export const PushValidationModal = {
   cnxmlErrorMsg: 'There are outstanding validation errors that must be resolved before pushing is allowed.',
   xmlErrorMsg: 'There are outstanding schema errors that must be resolved before pushing is allowed.'
 }
 
+export const getOpenDocuments = async (): Promise<Set<string>> => {
+  // People have asked for this for 6 years! https://github.com/Microsoft/vscode/issues/15178
```

Comment:
The goal of finding open documents was to (1) not open documents that are already open and (2) avoid automatically closing documents that were not automatically opened.

I tried a few things with the language server and could not find a good way to get a list of all open editor tabs. 

First I tried using  `ModelManager.openDocuments`. As you might imagine, however, this map only contains documents that the user has viewed (i.e. triggered the `onDidOpen` event).

Next I tried going a step lower and using the `TextDocuments.all` method. The list returned by this method, however, has the exact same information as `ModelManager.openDocuments`. 

In both cases, the problem is that the user might have tabs open that they have not viewed since opening their workspace. In this situation, those editor tabs (or their associated documents, rather) will not be present in those collections.